scarlet/frame.py

```diff
@@ -91,40 +93,45 @@ def psf(self):
 
     def get_pixel(self, sky_coord):
         """Get the pixel coordinate from a world coordinate
-        If there is no WCS associated with the `Scene`,
-        meaning the data frame and model frame are the same,
-        then this just returns `sky_coord`.
+
+        Parameters
+        ----------
+        sky_coord: tuple, array
+            Coordinates on the sky
         """
+        sky = np.array(sky_coord, dtype=np.float).reshape(-1, 2)
+
         if self.wcs is not None:
-            if self.wcs.naxis == 3:
-                coord = self.wcs.wcs_world2pix(*sky_coord, 0, 0)[:2]
-            elif self.wcs.naxis == 2:
-                coord = self.wcs.wcs_world2pix(*sky_coord, 0)
-            else:
-                raise ValueError(
-                    "Invalid number of wcs dimensions: {0}".format(self.wcs.naxis)
-                )
-            return tuple(int(c.item()) for c in coord)
+            pixel = np.array(self.wcs.world_to_pixel_values(sky)).reshape(-1, 2)
+            # y/x instead of x/y:
+            pixel = np.flip(pixel, axis=-1)
+        else:
+            pixel = sky
 
-        return tuple(int(coord) for coord in sky_coord)
+        if pixel.size == 2:  # only one coordinate pair
```

Comment:
Why would there be more than one pair? Also, if there is one pair, won't the size be 2?

Reply:
This method now allows passing (N,2) arrays so the N transformations are done in one go.